lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,326 @@
+        else:
+            deploy_content_standalone = \
+                {'operation': 'deploy',
+                 'address': [{'deployment': deployment}],
+                 'json.pretty': 1}
+            return get_response(deploy_content_standalone)
+    else:
+        module.fail_json(msg='Error while performing wildfly operation',
+                 details=info['msg'])
+
+def get_response(request_data):
+    headers = {'Content-Type': 'application/json'}
+    response = {}
+    info = {}
+    controller_address = module.params['controller_address']
+    try:
+        response, info = fetch_url(
+            module,
+            controller_address + '/management',
+            data=json.dumps(request_data),
+            headers=headers,
+            method='POST',
+        )
+    except urllib_error.HTTPError:
+        e = get_exception()
+        module.fail_json(msg='Error while performing wildfly operation',
+                 details=e.message)
+    
+    return response, info
+
+
+def is_deployed(params):
+    deployment = params['deployment']
+    request_data = {'operation': 'read-resource',
+                    'address': ['deployment', deployment],
+                    'json.pretty': 1}
+    return get_response(request_data)
+
+def is_deployed_domain(params):
+    deployment = params['deployment']
+    server_groups = params['server_groups']
+    request_data = \
+        {'operation': 'read-resource',
+         'address': [{'server-group': server_groups},
+                     {'deployment': deployment}],
+         'json.pretty': 1}
+    return get_response(request_data)
+
+
+def deploy_disable(result, params):
+    if result.get('outcome') == 'failed':
+        return False
+    elif result.get('outcome') == 'success':
+    	if not module.check_mode:
+            undeploy_result, info = undeploy(params)
+            if info['status'] == 200:
+                return True
+            else:
+                module.fail_json(msg='Undeploy failed', details=info['msg'])
+        return True
+
+
+def deploy_enable(result, params):
+    if result.get('outcome') == 'failed':
+    	if not module.check_mode:
+            deployment_result, info = deploy(params)
+            if info['status'] == 200:
+                return True
+            else:
+                module.fail_json(msg='Deployment failed', details=info['msg'])
+        else:
+            return True
+    elif result.get('outcome') == 'success':
+    	if not module.check_mode:
+            undeploy_result, info = undeploy(params)
+            if info['status'] == 200:
+                deployment_result, info = deploy(params)
+                if info['status'] == 200:
+                    return True
+                else:
+                    module.fail_json(msg='Deployment failed', details=info['msg'])
+        else:
+        	return True
+    else:
+        module.fail_json(msg='Deployment failed', details=info['msg'])
+
+
+def main():
+
+    global module
```

Comment:
If possible, do not use global variables. Pass the module to the function as a parameter if needed.